FinSearchUnified/tests/Components/ProductStream/CriteriaFactoryTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace FinSearchUnified\tests\Components\ProductStream;
+
+use FinSearchUnified\Constants;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class CriteriaFactoryTest extends TestCase
+{
+    /**
+     * Provider for test cases to test CriteriaFactory
+     *
+     * @return array
+     */
+    public function providers()
```

Comment:
Rename this to something more meaningful e.g. `shopSearchSwitchProvider` or similar.